add engine and storage faq (#797)
taptap/tds-docs: i18n/en/docusaurus-plugin-content-docs/current/sdk/engine/faq.mdx

You can use the interface for uploading files provided by the SDK,
though we generally recommend that you upload files with client SDKs without having the files go through Cloud Engine. This helps you reduce the unnecessary traffic generated.

> ## When using file upload in Cloud Engine, the error message is: failed to upload file to qiniu bcz current file has not data stream。

Reviewer: ```suggestion
## I get the error "failed to upload file to qiniu bcz current file has not data stream" when uploading files in Cloud Engine.
```

> LCFile internally uses a local cache (which requires a local directory) to hold the binary data before uploading it to the file storage service. Inside the cloud engine it is not possible to create such a cache directory or file due to insufficient permissions, resulting in errors like `failed to upload file to qiniu bcz current file has not data stream`.

Reviewer: ```suggestion
LCFile internally uses a local cache (which requires a local directory) to hold the binary data before uploading it to the file storage service. Inside the Cloud Engine it is not possible to create such a cache directory or file due to insufficient permissions, resulting in errors like `failed to upload file to qiniu bcz current file has not data stream`.
```

> We do not recommend using file services in the cloud engine for several reasons:

Reviewer: ```suggestion
We do not recommend using file services in the Cloud Engine for several reasons:
```

> 1. The permissions issue mentioned above will occur. This makes it impossible to create cache directories and temporary files in the cloud engine.

Reviewer: ```suggestion
1. The permission issue mentioned above will occur. This makes it impossible to create cache directories and temporary files in the Cloud Engine.
```

> 2. All file data goes to the server first, and then the API is called to push it to the file storage provider, which leads to low efficiency and high network cost for the user (the cloud engine also needs to pay for more than the standard traffic cost).

Reviewer: ```suggestion
2. All file data goes to the server first, and then the API is called to push it to the file storage provider, which leads to low efficiency and high network cost for the user (you will need to pay for the excessive traffic usage with Cloud Engine).
```

> It is recommended to use the SDK to upload the file, and then give the result (e.g. the object id or url of the file) to the server (which is the cloud engine code). Uploading on the client side will use the file storage provider's CDN link (nearby uploads), which makes the user experience faster and saves the cloud engine's traffic costs.

Reviewer: ```suggestion
It is recommended to use the SDK to upload the file, and then give the result (e.g. the object id or url of the file) to the server (the code for Cloud Engine). Uploading on the client side will use the file storage provider's CDN (servers close to the user), which makes the user experience faster and saves the traffic costs of Cloud Engine.
```
